lib/cocoapods/target/pod_target.rb

```diff
@@ -166,7 +165,24 @@ def spec_consumers
     def uses_swift?
       return @uses_swift if defined? @uses_swift
       @uses_swift = begin
-        file_accessors.any? do |file_accessor|
+        file_accessors.reject { |a| a.spec.test_specification? }.any? do |file_accessor|
+          file_accessor.source_files.any? { |sf| sf.extname == '.swift' }
+        end
+      end
+    end
+
+    # Checks whether a particular test type uses Swift or not.
+    #
+    # @param  [Symbol] test_type
+    #         The test type to check whether it uses Swift or not.
+    #
+    # @return [Boolean] Whether the target uses Swift code in the specified test type.
+    #
+    def uses_swift_for_test_type?(test_type)
```

Comment:
this should be deleted once we do a 1.7.0 perhaps where we actually create a separate instance of `PodTarget` for test specs instead of grouping everything together.

Reply:
But for now it will do.